projects/plugins/jetpack/modules/search/class-jetpack-instant-search.php

```diff
@@ -43,6 +78,10 @@ public function init_hooks() {
 			add_action( 'wp_footer', array( $this, 'print_instant_search_sidebar' ) );
 		} else {
 			add_action( 'update_option', array( $this, 'track_widget_updates' ), 10, 3 );
+			// The priority has to be lower than 10 to run before _wp_sidebars_changed.
+			// Which migrates widgets from old theme to the new one.
+			add_action( 'after_switch_theme', array( $this, 'save_old_sidebars_widgets' ), 5 );
```

Comment:
The [official docs](https://developer.wordpress.org/reference/hooks/after_switch_theme/) indicate that this action is fired multiple times. Is this desirable? Would using [`switch_theme`](https://developer.wordpress.org/reference/hooks/switch_theme/) be more appropriate here?

Reply:
Yep it might fire multiple times, it doesn't matter. If it fired once already, the process in `remove_wp_migrated_widgets` wouldn't run as the jetpack sidebar have the same size.

`switch_theme` is fired in a separate request before the one  that fires `after_switch_theme`, so the mechanism in the PR wouldn't work.